backend/app/infrastructure/external_api/bitquery/blocks_reward.py

```diff
@@ -0,0 +1,43 @@
+import requests
+
+import app as app_module
+from app.constants import BITQUERY_API
+from app.exceptions import ExternalApiException
+from app.extensions import w3
+from app.infrastructure.external_api.bitquery.req_producer import (
+    produce_payload,
+    BitQueryActions,
+    get_bitquery_header,
+)
+
+
+def accumulate_blocks_reward_wei(blocks: list) -> int:
+    blocks_reward_gwei = 0.0
+    for block in blocks:
+        blocks_reward_gwei += float(block["reward"])
+
+    return int(w3.to_wei(blocks_reward_gwei, "gwei"))
```

Comment:
I can see a few issues here:
- if the values in tests are taken from the API (0.024473700594149782), it means the unit should be "ether", not "gwei". The proper result in a test should be 17-digits long
- if we add such long floats we loose some precision at the end - 0.0779027949198489 float vs 0.077902794919848902 Decimal
- I'm worried about the performance of this solution - there will be a lot of blocks and python loops are extremely slow. I think it's better to use `pandas` as we did for withdrawals and MEVs
- this shouldn't be responsibility of the file to sum it up, but rather `app/modules/staking/proceeds/core.py`

Reply:
Totally agree with everything, messed up with the units - thank you for catching it 🙏